packages/backend-storage/src/factory.ts

```diff
@@ -46,7 +55,139 @@ class AmplifyStorageFactory
         getInstanceProps
       );
     }
-    return constructContainer.getOrCompute(this.generator) as AmplifyStorage;
+    const amplifyStorage = constructContainer.getOrCompute(
+      this.generator
+    ) as AmplifyStorage;
+
+    const firstStorage = Stack.of(amplifyStorage).node.children.filter(
+      (el) => el instanceof AmplifyStorage
+    )[0] as AmplifyStorage;
+
+    /*
+     * only call Aspects once,
+     * otherwise there will be the an error -
+     * "there is already a construct with name 'storageRegion'"
+     */
+    if (firstStorage.name === this.props.name) {
+      Aspects.of(Stack.of(amplifyStorage)).add(
```

Comment:
Regarding. `Stack.of(amplifyStorage)` .

I was chatting with @josefaidt about necessity to place every `defineFunction` in separate stack.
It may happen that we'll want same behavior for `defineStorage` as opt in feature.
If this happens. Will this algorithm work if we do `Stack.of(amplifyStorage)`, then traverse parent nodes to find root and place aspect there?

Reply:
That's a good catch. I need to investigate on that

- [ ]

Reply:
Yea, we should be able to use the parent stack

![image](https://github.com/user-attachments/assets/70c22820-3320-4f47-a4f2-717bd6977d5a)